### PR TITLE
Initial continuous integration with Travis CI

### PR DESCRIPTION
**Pull request by metachris**
Proposal of an initial continuous integration setup using [Travis CI](https://travis-ci.org) (see also #72).

If merged and [enabled at Travis CI](https://docs.travis-ci.com/user/getting-started/#To-get-started-with-Travis-CI), code pushes to GitHub will automatically build the code and run a number of tests to verify the neo-cli functionality. 

* The steps executed by Travis CI are defined in `.travis.yml`
* The proposed setup uses a Docker image based on the [microsoft/dotnet:2.0-sdk](https://hub.docker.com/r/microsoft/dotnet/) base image to build the code
* Currently 2 test stages are implemented:
  * **neo-cli tests using [expect](https://linux.die.net/man/1/expect)**: these tests interacts with neo-cli like any normal user, send commands and wait for correct output. This could catch errors like #70. Note: if output of neo-cli commands that are tested with `expect` change, the `test-neo-cli.expect` script will also need to be updated accordingly.
  * **JSON-RPC tests**: neo-cli is started in the background with `dotnet neo-cli.dll --rpc`, and the JSON-RPC api is tested with `curl` commands. Currently one json-rpc test is implemented: querying `totalSupply` of the RPX token. This can catch errors such as #71.

Example output of Travis CI: https://travis-ci.org/metachris/neo-cli/builds/322348356

This PR is an initial CI setup which can easily be extended with more tests in the future.

Looking forward to hearing your thoughts, and whether you are interested in all of those tests or just a part of it, or something completely different. I'd also like to know how you want to name the directory which contains the continuous integration scripts (in this proposal it's simply `/ci`).

### CODE REVIEW

File: ci/run-tests-in-docker.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#
+set -e
+
+cd /opt/neo-cli
+
+# Run tests with expect
+expect /opt/ci/test-neo-cli.expect
+
+# Start neo-cli in background for additional JSON-RPC tests
+screen -dmS node1 bash -c "dotnet neo-cli.dll --rpc"
```
Comment (AshRolls) [inline ci/run-tests-in-docker.sh]: We should make sure this screen is closed after tests have run

File: ci/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+This are files for the continuous integration with Travis CI.
```
Comment (AshRolls) [inline ci/README.md]: 'This' should be 'These'

File: ci/run-tests-in-docker.sh (hunk 1 above)
Comment (metachris) [reply ci/run-tests-in-docker.sh]: The Docker container is killed after the tests, so it wouldn't matter.

File: ci/README.md (hunk 2 above)
Comment (metachris) [reply ci/README.md]: fixed

**Comment by AshRolls** [comment]
I like these changes, they will add more robustness to this repo. Given the amount of functionality exposed through `neo-cli` for the main `neo` repo, we should try and restrict the testing to general `neo-cli` rather than trying to stretch tests to cover all `neo` functionality. 

`/ci` is fine for folder name

**Comment by erikzhang** [comment]
I have set it up. So how do I test it?

**Comment by metachris** [comment]
To run it locally, you'll need to have [Docker CE](https://docs.docker.com/engine/installation/) installed. Then just run `./ci/build-and-test.sh`.

If you want to set it up with this GitHub repository, enable this repository at Travis CI ([see this guide](https://docs.travis-ci.com/user/getting-started/#To-get-started-with-Travis-CI)), and merge this PR. On all code pushes (including the merge commit), the build at Travis is triggered. [This](https://travis-ci.org/metachris/neo-cli/builds/322348356) is how an example output at Travis looks like.

**Comment by erikzhang** [comment]
So, anyone has tested it locally?

**Comment by metachris** [comment]
I have, and it works for me (OSX). Also works when executed by Travis-CI which uses Ubuntu Linux to run the builds.

**Comment by metachris** [comment]
Works! You can see the output of the builds here: https://travis-ci.org/neo-project/neo-cli

If you want to add a travis-ci build status badge for the master branch like this: [Build Status](https://travis-ci.org/neo-project/neo-cli), you can use this code for the readme (markdown format): 

    [Build Status](https://travis-ci.org/neo-project/neo-cli)

**Pull request merged**